src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -1676,43 +1677,38 @@ private Expression ParseNormalMethodInvocation(Type type, Expression instance, i
 		//    return Expression.Call(typeof(Enumerable), signature.Name, typeArgs, args);
 		//}
 
-		private static Expression ParseDynamicProperty(Type type, Expression instance, string propertyOrFieldName)
+		/// <summary>
+		/// Returns null if <paramref name="t"/> is an Array type.  Needed because the <seealso cref="Microsoft.CSharp.RuntimeBinder.Binder"/> lookup methods fail with a <seealso cref="InvalidCastException"/> if the array type is used.
+		/// Everything still miraculously works on the array if null is given for the type.
+		/// </summary>
+		/// <param name="t"></param>
+		/// <returns></returns>
+		private static Type RemoveArrayType(Type t)
 		{
-			var binder = Microsoft.CSharp.RuntimeBinder.Binder.GetMember(
-				Microsoft.CSharp.RuntimeBinder.CSharpBinderFlags.None,
-				propertyOrFieldName,
-				type,
-				new[] { Microsoft.CSharp.RuntimeBinder.CSharpArgumentInfo.Create(Microsoft.CSharp.RuntimeBinder.CSharpArgumentInfoFlags.None, null) }
-			);
+			if (t == null || t.IsArray)
+			{
+				return null;
```

Comment:
If it works with null, it doesn't really matter, but maybe the binder requires the array's element type?

```c#
return t.GetElementType();
```

Reply:
It seems like it works just fine whether we return null or the element type.  This might be a detail that will require a very specific edge case for someone to encounter that isn't already covered by the unit tests to figure out which one is needed.